app/models/concerns/with_submissions.rb

```diff
@@ -22,7 +22,7 @@ def submitted_by?(user)
   end
 
   def last_submission(user)
-    submissions_for(user).last
+    user.exercise_progress_for(self).try(&:last_submission)
```

Comment:
In this case it's like an elvis operator: it sends the given message only if receptor is non null